core/src/test/java/org/pentaho/cassandra/driver/datastax/DriverCQLRowHandlerTest.java

```diff
@@ -215,6 +218,53 @@ public boolean matches( Object argument ) {
     } ) );
   }
 
+  @Test
```

Comment:
Wouldn't it be sufficient to mock a ValueMetaInterface (for the Date and Timestamp cases) and a Row and then unit test the ValueMetaInterface .readValue method directly? Perhaps not, but if so, I think the test could by much simpler.

Reply:
My only issue with this currently is that readValue is currently private and so I can't mock it as is. I could make this method static no problem so I don't think there's any security issues with also making this public. So I think making it public should be okay and it could simplify others tests as well, just debating it currently.